src/actions/coffee.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import type {Coffee} from '../database/coffee';
+
+// coffee actions
+//
+// for now these just gobble up the whole Coffee object rather than
+// being more lean about the data it uses.
+
+// because firebase doesn't attach the key that push() generates
+// to the coffee object, we need to add the id separately.
```

Comment:
this is a sort of non-obvious point. in pretty much *every other database in the world*, it is standard to attach the `id` or `key` directly to each entity in the database